### PR TITLE
feat(exporter): add Flow component for agent Integration

### LINKED ISSUE #5021: Support a Flow equivalent to the agent integration in static mode
### Request

Add some kind of Flow component for easily self-collecting agent metrics in Flow mode.

### Use case

Feature parity with static mode, where the agent integration is commonly used on bare metal installs.

### PR DESCRIPTION
**Pull request by hainenber**
#### PR Description

Add a Flow component, equivalent to agent's integration in static mode

#### Which issue(s) this PR fixes/closes

Close #5021 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- ~~[ ] Tests updated~~
- ~~[ ] Config converters updated~~

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.agent.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.agent/
+title: prometheus.exporter.agen
+---
+
+# prometheus.exporter.agent
+The `prometheus.exporter.agent` component collects and expose metrics about the agent itself.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.agent.md:7]:
```suggestion
The `prometheus.exporter.agent` component collects and exposes metrics about the agent itself.
```

**Review by clayton-cornell — COMMENTED** [review]
Minor tweak for the docs

File: docs/sources/flow/reference/components/prometheus.exporter.agent.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.agent/
+title: prometheus.exporter.agen
```
Comment (marctc) [inline docs/sources/flow/reference/components/prometheus.exporter.agent.md:3]:
```suggestion
title: prometheus.exporter.agent
```

**Comment by captncraig** [comment]
I think this may be more complicated than is needed for this to work. The common prometheus.exporter.* code is useful for proxying static mode integrations, but in this case we don't really need the complexity that adds. We really just need to export a list of arguments including the agent's existing `/metrics` endpoint. The only complication is using the `agent.internal` address instead of `localhost`. I don't know what the existing integration really gives us there, other than another package.

**Comment by tpaschalis** [comment]
@captncraig could you expand on how this alternative could look like? It would help move this PR ahead with a different direction.

File: component/prometheus/exporter/agent/agent.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/grafana/agent/pkg/integrations/agent"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "prometheus.exporter.agent",
+		Args:          Arguments{},
+		Exports:       exporter.Exports{},
+		Singleton:     true,
```
Comment (captncraig) [inline component/prometheus/exporter/agent/agent.go:15]: We should remove this singleton flag. It is going to be going away soon, and there isn't a hard technical reason you can't have 2 of these (even if it is silly :))

**Comment by captncraig** [comment]
I tested this, and it works fine. Any simplifications I want can be addressed in the future. The only change I want is to remove the singleton flag, and fix the docs to reflect that.

Once that is done and the conflicts fixed, I am happy to merge this.

**Review by clayton-cornell — COMMENTED** [review]
No changes for docs. OK as-is.

File: component/prometheus/exporter/agent/agent.go (hunk 3 above)
Comment (hainenber) [reply component/prometheus/exporter/agent/agent.go]: Singleton removed. Thanks for the context! 😄

**Review by captncraig — APPROVED** [review]

**Pull request merged**